cli/cli.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+
+"""
+CLI for interfacing with deployed analysis runner.
+See README.md for more information.
+"""
+import re
+import logging
+import subprocess
+
+import click
+import requests
+import google.auth
+import google.auth.transport.requests
+
+logging.basicConfig(level='INFO')
+
+BRANCH = 'main'
+DEFAULT_SERVER_LOOKUP = (
+    f'https://raw.githubusercontent.com/'
+    f'populationgenomics/analysis-runner/{BRANCH}/cli/servermap.json'
+)
+# Make this None if you want to support all organisations
+SUPPORTED_ORGANISATIONS = {'populationgenomics'}
+
+
+@click.command(
+    help='CLI for the analysis runner - a CPG service for running analysis from '
+    'some GitHub repository (at a specific commit). The parameters are used to form a '
+    'POST request, sent to a server based on the --dataset parameter you provide.'
+)
+@click.option(
+    '--dataset',
+    required=True,
+    help='The dataset name, which determines which '
+    'analysis-runner server to send the request to',
+)
+@click.option(
+    '--output-dir',
+    '-o',
+    required=True,
+    help='The output directory of the run, MUST start with gs://',
+)
+@click.option(
+    '--repository',
+    '--repo',
+    help='The URI of the repository to run, must be approved by the appropriate server.'
+    ' Default behaviour is to find the repository of the current working'
+    ' directory with `git remote get-url origin`',
+)
+@click.option(
+    '--commit',
+    '--hash',
+    help='The hash of commit to run, default behaviour is to ' 'use the current commit',
+)
+@click.option(
+    '--description',
+    required=True,
+    help='Description of job, otherwise defaults to: "$USER FROM LOCAL: $REPO@$COMMIT"',
+)
+@click.argument('script', nargs=-1)
+def main(dataset, output_dir, script, description, commit=None, repository=None):
+    """
+    Main function that drives the CLI.
+    The parameters are provided automatically by @click.
+    """
+
+    if repository is not None and commit is None:
+        raise Exception(
+            "You must supply the '--commit <SHA>' parameter "
+            "when specifying the '--repository'"
+        )
+
+    _repository = repository or _get_default_remote()
+    _commit_ref = commit or _get_commit_ref_of_current_repository(
+        used_custom_repository=repository is not None
+    )
+    _url = _get_url_from_dataset(dataset)
+    _token = _get_google_auth_token()
+    _script = script or ['main.py']
```

Comment:
We should grab the current working directory here. I.e. if my repository root is at `/my/repo` and I'm currently in `/my/repo/analysis/project` and pass `main.py` as the script, the command should become `analysis/project/main.py`.

Reply:
Sweet! The script does something like this now:

```python
if _script[0]:
    # Make the first argument (the script name) relative
    # to the git root and current directory
    _script[0] = _get_relative_script_path_from_git_root(_script[0])
```